source/module_beyonddft/dm_trans/dm_trans_parallel.cpp

```diff
@@ -0,0 +1,214 @@
+#ifdef __MPI
+#include "dm_trans.h"
+#include "module_base/scalapack_connector.h"
+#include "module_base/tool_title.h"
+#include "module_beyonddft/utils/lr_util.h"
+namespace hamilt
+{
+
+// output: col first, consistent with blas
+//  c: nao*nbands in para2d, nbands*nao in psi  (row-para and constructed: nao)
+//  X: nvirt*nocc in para2d, nocc*nvirt in psi (row-para and constructed: nvirt)
+template <>
+std::vector<container::Tensor> cal_dm_trans_pblas(const psi::Psi<double>& X_istate,
+                                                  const Parallel_2D& px,
+                                                  const psi::Psi<double>& c,
+                                                  const Parallel_2D& pc,
+                                                  int naos,
+                                                  int nocc,
+                                                  int nvirt,
+                                                  Parallel_2D& pmat,
+                                                  const bool renorm_k,
+                                                  const int nspin)
+{
+    ModuleBase::TITLE("hamilt_lrtd", "cal_dm_trans_pblas");
+    assert(px.comm_2D == pc.comm_2D);
+    assert(px.blacs_ctxt == pc.blacs_ctxt);
+
+    if (pmat.comm_2D != px.comm_2D || pmat.blacs_ctxt != px.blacs_ctxt)
+        LR_Util::setup_2d_division(pmat, px.get_block_size(), naos, naos, px.comm_2D, px.blacs_ctxt);
+    else
+        assert(pmat.get_local_size() > 0);
+
+    int nks = c.get_nk();
+    assert(nks == X_istate.get_nk());
+
+    std::vector<container::Tensor> dm_trans(
+        nks,
+        container::Tensor(DAT::DT_DOUBLE, DEV::CpuDevice, {pmat.get_col_size(), pmat.get_row_size()}));
```

Comment:
what are DAT and DEV, where are they defined?

Reply:
In `lr_util.h`: 
```
using DAT = container::DataType;
using DEV = container::DeviceType;
```